app/lib/database/repositories/index.js

```diff
@@ -24,12 +23,37 @@ const Repository = require('./Repository.js');
  */
 const specificallyDefinedRepositories = {};
 
-module.exports = (sequelize) => {
-    const modelName2Repository = Object.entries(modelsCreator(sequelize)).map(([modelName, model]) => {
-        return [modelName + 'Repository', modelName in specificallyDefinedRepositories ? new specificallyDefinedRepositories[modelName](model) : new Repository(model)];
-    });
+/**
+ * @see specificallyDefinedRepositories
+ * It checks if specificallyDefinedRepositories is correct, if not, it throws an error.
+ * @param {Object<string, Sequelize.Model>} models 
+ */
+const validateSpecificRepositoriesConfiguration = (models) => {
+    Object.entries(specificallyDefinedRepositories).forEach(([modelName, RepoClass]) => { 
+        if (! (modelName in models) || (! (RepoClass.prototype instanceof Repository))) {
+            throw Error({message: `Incorrect configuration of specificallyDefinedRepositories for modelName: <${modelName}>`})
+        }
+    })
+}
+
+/**
+ * Instantiate sequelize models repositories according to repositiry pattern.
+ * @param {Object<string, Sequelize.Model>} models dict: modelName -> sequelize model, @see specificallyDefinedRepositories
+ * @returns {Object<string, Repository>} dict: repositoryName -> repository instance per one model, (repositoryName = modelName + 'Repository')
+ */
+const repositoriesFactory = (models) => {
+    validateSpecificRepositoriesConfiguration(models);
+
+    const modelName2Repository = Object.entries(models).map(([modelName, model]) =>
```

Comment:
I would not use 2 in the name and prefer `to`, it makes the name more obvious for non-native speakers